File: @LV1/1.7/jewelrybox/jewelrybox.py
def find_height(a, b):
    return ((a+b) - ((a+b)**2 - 3*a*b)**0.5)/6


def max_volumn(a, b):
    m_h = find_height(a, b)
    return m_h * (a-2*m_h)*(b-2*m_h)


n = int(input())
inp = [[int(i) for i in input().split(' ')] for _ in range(n)]
[print(max_volumn(*i)) for i in inp]

File: @LV1/1.7/jewelrybox/jewelrybox.rb
def find_height(a, b)
  ((a + b) - ((a + b)**2 - 3 * a * b)**0.5) / 6
end

def max_volumn(a, b)
  m_h = find_height(a, b)
  m_h * (a - 2 * m_h) * (b - 2 * m_h)
end

n = gets.to_i
inp = (1..n).map { gets.split.map(&:to_i) }
inp.map { |a, b| puts max_volumn(a, b) }
